File: Python/Anagrams.py
class Solution:
    # @param strs, a list of strings
    # @return a list of strings
    def anagrams(self, strs):
        d = {}
        for item in strs:
            ans = list(item)
            ans.sort()
            s = "".join(ans)
            if s not in d:
                d[s] = []
            d[s].append(item[:])
        seq = []
        for item in d:
            if len(d[item]) >= 2:
                seq += d[item]
        return seq

File: C++/Anagrams.cpp
class Solution {
public:
    vector<string> anagrams(vector<string>& strs) {
        map<string, vector<string> >hash;
        for (int i = 0; i < strs.size(); i ++) {
            string s = strs[i];
            sort(s.begin(), s.end());
            if (hash.find(s) != hash.end()) {
                hash[s].push_back(strs[i]);
            } else {
                vector<string> temp;
                temp.push_back(strs[i]);
                hash.insert(make_pair(s, temp));
            }
        }
        vector<string> res;
        for (auto it = hash.begin(); it != hash.end(); it ++) {
            if ((*it).second.size() >= 2) {
                for (int i = 0; i < (*it).second.size(); i ++) {
                    res.push_back(((*it).second)[i]);
                }
            }
        }
        return res;
    }
};